stable/aws-cloudwatch-metrics/templates/_helpers.tpl

```diff
@@ -0,0 +1,62 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "aws-cloudwatch-metrics.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "aws-cloudwatch-metrics.fullname" -}}
+{{- if .Values.fullnameOverride -}}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "aws-cloudwatch-metrics.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "aws-cloudwatch-metrics.labels" -}}
+helm.sh/chart: {{ include "aws-cloudwatch-metrics.chart" . }}
+{{ include "aws-cloudwatch-metrics.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end -}}
+
+{{/*
+Selector labels
+*/}}
+{{- define "aws-cloudwatch-metrics.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "aws-cloudwatch-metrics.name" . }}
+{{- end -}}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "aws-cloudwatch-metrics.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create -}}
+    {{ default (include "aws-cloudwatch-metrics.fullname" .) .Values.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.serviceAccount.name }}
```

Comment:
And one other issue I see is that this is copying in a mistake that was fixed here: https://github.com/aws/eks-charts/commit/8c0db8a9e3845229f4d8806ccc498f2635ac8a29

Basically, you need to remove the `{{ if ` block around the `Values.serviceAccount.create` check above...

Reply:
oops :slightly_smiling_face:

Reply:
Hi. this service account name template is not used in your cluster role binding. There it just uses  {{ include "aws-cloudwatch-metrics.fullname" . }}. It should also use the template aws-cloudwatch-metrics.serviceAccountName in role binding I think.